.github/workflows/build.yml

```diff
@@ -0,0 +1,68 @@
+name: build
+
+on:
+  push:
+    branches: ["*"]
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version to build, e.g. 0.1.0"
+        type: string
+        required: false
+
+env:
+  PKG_NAME: "vault-servicenow-credential-resolver"
+
+jobs:
+  get-product-version:
+    runs-on: ubuntu-latest
+    outputs:
+      product-version: ${{ steps.get-product-version.outputs.product-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: get product version
+        id: get-product-version
+        run: |
+          VERSION="${{ github.event.inputs.version || '0.0.0-dev' }}"
+          echo "Using version ${VERSION}"
+          echo "::set-output name=product-version::${VERSION}"
+  generate-metadata-file:
+    needs: get-product-version
+    runs-on: ubuntu-latest
+    outputs:
+      filepath: ${{ steps.generate-metadata-file.outputs.filepath }}
+    steps:
+      - name: 'Checkout directory'
+        uses: actions/checkout@v2
+      - name: Generate metadata file
+        id: generate-metadata-file
+        uses: hashicorp/actions-generate-metadata@v1
+        with:
+          version: ${{ needs.get-product-version.outputs.product-version }}
+          product: ${{ env.PKG_NAME }}
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: metadata.json
+          path: ${{ steps.generate-metadata-file.outputs.filepath }}
+
+  build:
+    needs: get-product-version
+    runs-on: ubuntu-latest
+
+    name: Build Jar
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build
+        env:
+          VERSION: ${{ needs.get-product-version.outputs.product-version }}
+        run: |
+          mkdir out
+          ./gradlew jar
+          zip -r -j out/${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_all_all.zip build/libs/
```

Comment:
This may be my ignorance, but why do we need to ZIP the JAR file?

Reply:
Cool. It's fine, it just seems weird to ZIP a file that is already a ZIP, when the end user will probably just want the JAR anyway.

Reply:
Yeah it is a little bit odd, a bit like the `all_all` os/arch strings. The `all_all` string was to conform with an internal RFC to help facilitate consistent release tooling. The RFC doesn't address JAR files specifically so it might be worth checking. I'll follow up on this